src/NexusMods.App.UI/Helpers/DoubleClickHelper.cs

```diff
@@ -0,0 +1,117 @@
+using System.Reactive;
+using System.Reactive.Disposables;
+using System.Reactive.Subjects;
+using Avalonia.Controls;
+using Avalonia.Input;
+using Avalonia.Interactivity;
+using JetBrains.Annotations;
+
+namespace NexusMods.App.UI.Helpers;
+
+[PublicAPI]
+public static class DoubleClickHelper
+{
+    // NOTE(erri120): Windows uses 500ms (https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-setdoubleclicktime?redirectedfrom=MSDN#parameters)
+    // This can be fine-tuned later.
+    public static readonly TimeSpan DefaultTimeout = TimeSpan.FromMilliseconds(500);
+
+    /// <summary>
+    /// Registers a double click handler and returns an observable that triggers whenever
+    /// a double click occurs.
+    /// </summary>
+    public static IObservable<Unit> AddDoubleClickHandler(this Control control)
+    {
+        return AddDoubleClickHandler(control, InputElement.PointerPressedEvent);
+    }
+
+    /// <summary>
+    /// Registers a double click handler and returns an observable that triggers whenever
+    /// a double click occurs.
+    /// </summary>
+    public static IObservable<Unit> AddDoubleClickHandler<TRoutedEventArgs>(this Control control, RoutedEvent<TRoutedEventArgs> routedEvent)
+        where TRoutedEventArgs : RoutedEventArgs
+    {
+        return new DoubleClickSubject<TRoutedEventArgs>(control, routedEvent);
+    }
+
+    private class DoubleClickSubject<TRoutedEventArgs> : SubjectBase<Unit>
+        where TRoutedEventArgs : RoutedEventArgs
+    {
+        private bool _isDisposed;
+
+        private readonly List<IObserver<Unit>> _observers = new(capacity: 1);
+        private readonly IDisposable _handlerDisposable;
+
+        public DoubleClickSubject(Control control, RoutedEvent<TRoutedEventArgs> routedEvent)
+        {
+            _handlerDisposable = control.AddDisposableHandler(
+                routedEvent,
+                EventHandler,
+                routes: RoutingStrategies.Bubble,
+                handledEventsToo: true
```

Comment:
What are the use cases for handling already handled events?
I thought this would help prevent double clicks from propagating up from nested controls that have their own click handlers?

Reply:
The `PointerPressedEvent` is often already handled by something else. Think about the item selection in a list. The event propagating is done with the event routes, which I set to `Bubble`, meaning events from child controls will bubble up to the parent.

I'll add parameters to the API, so consumers can choose whether they want events from child controls to appear as well.

Reply:
Does item selection actually mark the click event as handled? That doesn't sound like a great idea, I didn't know it did that.
Api exposing this to users sounds good. 

If we want to add a double click handler to items in a DataGrid, but these items also have a toggle button, we would want double click to work except when pressing on the toggle button. If the first click is already eaten by Selection then we are a bit in trouble, as we would have to handle Handled events to avoid Selection eating the first click, but would end up also handling the Toggle button events that way.

Reply:
> Does item selection actually mark the click event as handled? That doesn't sound like a great idea, I didn't know it did that.

https://github.com/AvaloniaUI/Avalonia/blob/4fdf4fa36cfc9b1e353a4ca79e5c8e8137bd9ea2/src/Avalonia.Controls/ListBoxItem.cs#L68

> If we want to add a double click handler to items in a DataGrid, but these items also have a toggle button, we would want double click to work except when pressing on the toggle button.

Toggling the button is a single-click operation. Unless the user actually double-clicks, the double-click observable won't trigger.

Reply:
The toggle button is the child in this scenario, so it isn't a problem that we could set the event as handled, the problem is that we want to ignore events handled by the button.

If the user double clicks on a child button, like a three state checkbox or whatever, I wouldn't expect the parent control to behave as it if were double clicked, because the events should have been consumed/handled by the child control already.

But if we need to listen to Handled events as well due to Selection otherwise eating those, then we end up listening to the child button clicks as well. 

I think we might be able to resolve this by choosing exactly which control we attach the handler to, e.g. a border that is inside the DataGrid row or cell, rather than on the row or cell itself.